xbmc/input/InputCodingTableBaiduPY.cpp

```diff
@@ -121,6 +122,30 @@ std::vector<std::wstring> CInputCodingTableBaiduPY::GetResponse(int response)
   return words;
 }
 
+bool CInputCodingTableBaiduPY::Initialize()
+{
+  CSingleLock lock(m_CS);

```

Comment:
What are you protecting here? If it's `m_initialized` you also need to use the lock in `UnInitialize`.